plugin/trino-pinot/src/test/java/io/trino/plugin/pinot/TestPinotIntegrationSmokeTest.java

```diff
@@ -796,4 +861,264 @@ public void testLimitPushdown()
+
+        // Test aggregates with no grouping columns with a filter
+        assertThat(query("SELECT COUNT(*)," +
+                "  MIN(int_col), MAX(int_col)," +
+                "  MIN(long_col), MAX(long_col), AVG(long_col), SUM(long_col)," +
+                "  MIN(float_col), MAX(float_col), AVG(float_col), SUM(float_col)," +
+                "  MIN(double_col), MAX(double_col), AVG(double_col), SUM(double_col)" +
+                "  FROM " + ALL_TYPES_TABLE + " WHERE long_col < 4147483649"))
+                .isFullyPushedDown();
+
+        // Test aggregates with no grouping columns with a filter and limit
+        assertThat(query("SELECT COUNT(*)," +
+                "  MIN(int_col), MAX(int_col)," +
+                "  MIN(long_col), MAX(long_col), AVG(long_col), SUM(long_col)," +
+                "  MIN(float_col), MAX(float_col), AVG(float_col), SUM(float_col)," +
+                "  MIN(double_col), MAX(double_col), AVG(double_col), SUM(double_col)" +
+                "  FROM " + ALL_TYPES_TABLE + " WHERE long_col < 4147483649" +
+                "  LIMIT " + MAX_ROWS_PER_SPLIT_FOR_SEGMENT_QUERIES))
+                .isFullyPushedDown();
+
+        // Test aggregates with one grouping column
+        assertThat(query("SELECT bool_col, COUNT(*)," +
+                "  MIN(int_col), MAX(int_col)," +
+                "  MIN(long_col), MAX(long_col), AVG(long_col), SUM(long_col)," +
+                "  MIN(float_col), MAX(float_col), AVG(float_col), SUM(float_col)," +
+                "  MIN(double_col), MAX(double_col), AVG(double_col), SUM(double_col)" +
+                "  FROM " + ALL_TYPES_TABLE + " GROUP BY bool_col"))
+                .isFullyPushedDown();
+
+        // Test aggregates with one grouping column and a limit
+        assertThat(query("SELECT string_col, COUNT(*)," +
+                "  MIN(int_col), MAX(int_col)," +
+                "  MIN(long_col), MAX(long_col), AVG(long_col), SUM(long_col)," +
+                "  MIN(float_col), MAX(float_col), AVG(float_col), SUM(float_col)," +
+                "  MIN(double_col), MAX(double_col), AVG(double_col), SUM(double_col)" +
+                "  FROM " + ALL_TYPES_TABLE + " GROUP BY string_col" +
+                "  LIMIT " + MAX_ROWS_PER_SPLIT_FOR_SEGMENT_QUERIES))
+                .isFullyPushedDown();
+
+        // Test aggregates with one grouping column and a filter
+        assertThat(query("SELECT bool_col, COUNT(*)," +
+                "  MIN(int_col), MAX(int_col)," +
+                "  MIN(long_col), MAX(long_col), AVG(long_col), SUM(long_col)," +
+                "  MIN(float_col), MAX(float_col), AVG(float_col), SUM(float_col)," +
+                "  MIN(double_col), MAX(double_col), AVG(double_col), SUM(double_col)" +
+                "  FROM " + ALL_TYPES_TABLE + " WHERE long_col < 4147483649 GROUP BY bool_col"))
+                .isFullyPushedDown();
+
+        // Test aggregates with one grouping column, a filter and a limit
```

Comment:
https://github.com/trinodb/trino/pull/6069/files#r683972909

Reply:
Also, can we tests for MAX/MIN over text types? This is related to https://github.com/trinodb/trino/pull/8551.

Reply:
Pinot does not support min/max for varchar or boolean using this version of the pinot libraries (0.6.0). Will upgrade soon and retest. The min/max aggregations are not pushed down: see SUPPORTED_TYPES in the ImplementMinMax.

Reply:
Added some tests for mixed case values with distinct, count distinct and approx_distinct.